src/controllers/dynakube/activegate/internal/capability/reconciler_test.go

```diff
@@ -1,517 +1,360 @@
+		mockStatefulSetReconciler.AssertCalled(t, "Reconcile")
+		mockCustompropertiesReconciler.AssertCalled(t, "Reconcile")
+		require.NoError(t, err)
+
+		service := corev1.Service{}
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, &service)
+
+		assert.NotNil(t, service)
+		assert.NoError(t, err)
+	})
+	t.Run(`Service doesnt get created when missing capabilities`, func(t *testing.T) {
+		clt := createClient()
+		instance := createInstanceWithoutActiveGateService()
+		mockStatefulSetReconciler := getMockReconciler()
+		mockCustompropertiesReconciler := getMockReconciler()
+
+		r := NewReconciler(clt, capability.NewMultiCapability(instance), instance, mockStatefulSetReconciler, mockCustompropertiesReconciler)
+		verifyReconciler(t, r)
+
+		err := r.Reconcile()
+		mockStatefulSetReconciler.AssertCalled(t, "Reconcile")
+		mockCustompropertiesReconciler.AssertCalled(t, "Reconcile")
+		require.NoError(t, err)
+
+		service := corev1.Service{}
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, &service)
+
+		assert.Empty(t, service)
+		assert.Error(t, err)
+	})
+}
+
+func TestCreateOrUpdateService(t *testing.T) {
+	t.Run(`createOrUpdateService works`, func(t *testing.T) {
+		clt := createClient()
+		instance := createInstance()
+		mockStatefulSetReconciler := getMockReconciler()
+		mockCustompropertiesReconciler := getMockReconciler()
+
+		r := NewReconciler(clt, capability.NewMultiCapability(instance), instance, mockStatefulSetReconciler, mockCustompropertiesReconciler)
+		verifyReconciler(t, r)
+
+		service := &corev1.Service{}
+		err := r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, service)
+		require.Error(t, err)
+		assert.NotNil(t, service)
+
+		err = r.createOrUpdateService()
+		require.NoError(t, err)
+
+		service = &corev1.Service{}
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, service)
+		require.NoError(t, err)
+		assert.NotNil(t, service)
+	})
+	t.Run(`Update works for ports`, func(t *testing.T) {
+		clt := createClient()
+		instance := createInstance()
+		mockStatefulSetReconciler := getMockReconciler()
+		mockCustompropertiesReconciler := getMockReconciler()
+
+		r := NewReconciler(clt, capability.NewMultiCapability(instance), instance, mockStatefulSetReconciler, mockCustompropertiesReconciler)
+		verifyReconciler(t, r)
+
+		err := r.createOrUpdateService()
+		require.NoError(t, err)
+
+		service := &corev1.Service{}
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: r.dynakube.Name + "-" + r.capability.ShortName(), Namespace: r.dynakube.Namespace}, service)
+		require.NoError(t, err)
+		assert.NotNil(t, service)
+
+		tmpService := &corev1.Service{}
```

Comment:
```suggestion
		actualService := &corev1.Service{}
```
declare close to where it's used